tests/FrameCameraTests.cpp

```diff
@@ -39,118 +38,56 @@ class FramerParameterizedTest : public ::testing::TestWithParam<csm::ImageCoord>
           return sensorModel;
         else
           return nullptr;
-
-
       }
 
 
-
    virtual void SetUp() {
-
-
-      std::ifstream isdFile("data/simpleFramerISD.json");
-      json jsonIsd = json::parse(isdFile);
-      for (json::iterator it = jsonIsd.begin(); it != jsonIsd.end(); ++it) {
-         json jsonValue = it.value();
-         if (jsonValue.size() > 1) {
-            for (int i = 0; i < jsonValue.size(); i++) {
-               isd.addParam(it.key(), jsonValue[i].dump());
-            }
-         }
-         else {
-            isd.addParam(it.key(), jsonValue.dump());
-         }
-      }
-      isdFile.close();
-   }
+     isd.setFilename("data/simpleFramerISD.json");
+   };
 };
 
 
-
-
 class FrameIsdTest : public ::testing::Test {
-   protected:
-
-      csm::Isd isd;
-
-      void printIsd(csm::Isd &localIsd) {
-           multimap<string,string> isdmap= localIsd.parameters();
-           for (auto it = isdmap.begin(); it != isdmap.end();++it){
-
-                      cout << it->first << " : " << it->second << endl;
-           }
-       }
-      UsgsAstroFrameSensorModel* createModel(csm::Isd &modifiedIsd) {
-
-        UsgsAstroFramePlugin frameCameraPlugin;
-        csm::Model *model = frameCameraPlugin.constructModelFromISD(
+  protected:
+    csm::Isd isd;
+    void printIsd(csm::Isd &localIsd) {
+      multimap<string,string> isdmap= localIsd.parameters();
+      for (auto it = isdmap.begin(); it != isdmap.end();++it){
+        cout << it->first << " : " << it->second << endl;
+      }
+    }
+    UsgsAstroFrameSensorModel* createModel(csm::Isd &modifiedIsd) {
+      UsgsAstroFramePlugin frameCameraPlugin;
+      csm::Model *model = frameCameraPlugin.constructModelFromISD(
               modifiedIsd,"USGS_ASTRO_FRAME_SENSOR_MODEL");
-
-        UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
-
-        if (sensorModel)
-          return sensorModel;
-        else
-          return nullptr;
-
+      UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
+      if (sensorModel)
+        return sensorModel;
+      else
+        return nullptr;
       }
 
 
-
-   virtual void SetUp() {
-      std::ifstream isdFile("data/simpleFramerISD.json");
-      json jsonIsd = json::parse(isdFile);
-      for (json::iterator it = jsonIsd.begin(); it != jsonIsd.end(); ++it) {
-         json jsonValue = it.value();
-         if (jsonValue.size() > 1) {
-            for (int i = 0; i < jsonValue.size(); i++) {
-               isd.addParam(it.key(), jsonValue[i].dump());
-            }
-         }
-         else {
-            isd.addParam(it.key(), jsonValue.dump());
-         }
-      }
-      isdFile.close();
+    virtual void SetUp() {
+      isd.setFilename("data/simpleFramerISD.json");
```

Comment:
It's expected that the filename will be the image filename. It's not a functional difference, but I think it's a better test of expected functionality if this ends in `.img` or `.cub` or `.raw`

Reply:
:+1: